web/app/tournament/tourny.py

```diff
@@ -169,3 +172,93 @@ def select_player(tournament_id, tournament_player_id):
         # flash("Could not delete player: {{ tp.name }}")
         return redirect(url_for('.players_index',
             tournament_id=tournament_id))
+
+
+@tournament.route('/<tournament_id>/matches/', methods=['GET', 'POST'])
+def matches_index(tournament_id):
+    if request.method == "GET":
+        matches = Match.query.filter_by(tournament_id=tournament_id)
+        return render_template('tournament_match_index.html',
+        tournament_id=tournament_id,
+        matches=matches)
+
+    if request.method == 'POST':
+        players = TournamentPlayer.query.filter_by(tournament_id=tournament_id)
+        players_list = list(players)
+        shuffle(players_list)
+        while len(players_list) > 0:
+            if len(players_list) % 2 == 0:
+                match = Match(player_1_id = players_list.pop().id,
+                              player_2_id = players_list.pop().id,
+                              tournament_id = tournament_id)
+                db.session.add(match)
+                db.session.commit()
+                match = Match.query.all()[-1]
+                match.player_1_name = players.filter_by(
+                    id=match.player_1_id).first().name
+                match.player_2_name = players.filter_by(
+                    id=match.player_2_id).first().name
+                db.session.add(match)
+                db.session.commit()
+            else:
+                match = Match(player_1_id = players_list.pop().id,
+                              player_2_id = -1,
+                              tournament_id = tournament_id)
+                db.session.add(match)
+                db.session.commit()
+                match = Match.query.all()[-1]
+                match.player_1_name = players.filter_by(
+                    id=match.player_1_id).first().name
+                match.player_2_name = "BYE"
+                db.session.add(match)
+                db.session.commit()
+        return redirect(url_for('.matches_index',
+        tournament_id=tournament_id))
+
+
+@tournament.route('/<tournament_id>/match/<match_id>/', methods=['POST'])
+def match(tournament_id, match_id):
```

Comment:
nit: Some method docstrings would be nice here: https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments